samples/13.core-bot/src/main/java/com/microsoft/bot/sample/core/DialogBot.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.sample.core;
+
+import com.microsoft.bot.builder.*;
+import com.microsoft.bot.dialogs.Dialog;
+
+import java.util.concurrent.CompletableFuture;
+
+// This IBot implementation can run any type of Dialog. The use of type parameterization is to allows multiple different bots
+// to be run at different endpoints within the same project...
+public class DialogBot<T extends Dialog> extends ActivityHandler {
+    protected Dialog dialog;
+    protected BotState conversationState;
+    protected BotState userState;
+
+    public DialogBot(ConversationState conversationState, UserState userState, T dialog) {
```

Comment:
```suggestion
    public DialogBot(ConversationState withConversationState, UserState withUserState, T withDialog) {
```

Checkstyle: the parameters can't contain the same name as the object's variables

Reply:
Good point. I'm worried I think we didn't do this on the qna sample bot

Reply:
See [qna](https://github.com/microsoft/botbuilder-java/blob/main/samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/QnABot.java#L19)